### ISSUE TITLE
Switch from boost to ext and ql includes

### ISSUE THREAD

**Issue by eltoder** [OpenSourceRisk/Engine#151]
Upstream QuantLib is moving from boost classes to std classes for shared_ptr, optional, etc. To ease the transition they added the ext:: namespace which has aliases to either boost or std classes depending on the config options. Are you planning to update your code to use the ext namespace to make it closer to the upstream? This is mostly a search and replace exercise but there are a few thousand hits.

**Comment by eltoder** [comment]
@pcaspers could you opine please?

**Comment by pcaspers** [comment]
@eltoder Yes this is already on our radar. I expect this to be done in the next release.

**Comment by QuaternionRisk** [comment]
We did not manage to get this done for the 10th release, unfortunately. We will keep it on our roadmap for the next one (end of September) .
Regards, Roland

**Comment by UnitedMarsupials** [comment]
Here is, what I had to do to achieve this. Perl or sed can be used to do this with a single command:
1. Replace `boost::(make_shared|shared_ptr|optional|dynamic_pointer_cast|none|static_pointer_cast)` with `QuantLib::ext::\1` throughout.
2. Subsequently replace `ext::none` with `ext::nullopt`.

If only figuring out the `fairSpreadClean()` added to the fork were as easy :-/
